### PR TITLE
Add support for "compare_log" in GreenTea test_spec.json

### PR DESCRIPTION
**Pull request by jamesbeyond**
### Description

This change add the support for "compare_log" key in the test_spec.json file
This change would allow mbed-os examples be tested by passing a customized test_spec.json file.
And utilized HTrun [log compare](https://github.com/ARMmbed/mbed-os-tools/tree/master/packages/mbed-host-tests#testing-mbed-os-examples) function.

The normal Green Tea test functions will not be impacted.

Detailed changes:
 * in TestBinary class add compare_log member to the constructor
 * in Greentea passing the compare_log to HTrun command line

### Pull request type

    [ ] Fix
    [ ] Refactor
    [ ] Target update
    [x] Functionality change
    [ ] Docs update
    [ ] Test update
    [ ] Breaking change

### CODE REVIEW

File: src/mbed_os_tools/test/tests_spec.py [hunk 1]
```diff
@@ -48,13 +49,20 @@ def __init__(self, path, binary_type):
...
         self.__flash_method = binary_type
+        self.__comp_log = compare_log
 
     def get_path(self):
         """
         Gives binary path.
         :return:
         """
         return self.__path
+    def get_comp_log(self):
```
Comment (bridadan) [inline src/mbed_os_tools/test/tests_spec.py]: I prefer not to abbreviate stuff if the full phrase is still quite short (just a difference of three letters):
```suggestion
    def get_compare_log(self):
```

**Comment by bridadan** [comment]
I'm not sure why the "Test WindowsPython27" build is failing to install, I think that's an issue on master I need to sort out. However there is a new linting error that was introduced and caught by the "Test LinuxPython37" job, could you please address the following lint errors?

```
./src/mbed_os_tools/test/tests_spec.py:60:5: E301 expected 1 blank line, found 0
./src/mbed_os_tools/test/tests_spec.py:120:89: E501 line too long (99 > 88 characters)
./src/mbed_os_tools/test/tests_spec.py:131:89: E501 line too long (97 > 88 characters)
```

**Comment by bridadan** [comment]
Another question: does this work fine with older test_spec.json files that do not have the `compare_log` key? Greentea should be able to handle these older files for previous releases.

File: src/mbed_os_tools/test/tests_spec.py (hunk 1 above)
Comment (jamesbeyond) [reply src/mbed_os_tools/test/tests_spec.py]: Fixed

**Comment by jamesbeyond** [comment]
Did aware that you got azure pipeline checking for the python format, which is very nice!
I fixed the issues you mentioned.
The changes should not have any impact to those not having `compare_log` key, because the default value for the `compare_log` is `None` anyway. I tested it on my local machine. 
Also the GreenTea test will not generate `compare_log` key by any means, this will only be used for example testing

**Comment by jamesbeyond** [comment]
Oh, seems it exposed some more format issues, I'll look into that

**Comment by jamesbeyond** [comment]
Any idea what are these errors about? @bridadan

**Comment by bridadan** [comment]
There seems to be quite a few Python requirements issues popping up, this should not be related to your changes. I'll keep you posted as I look into this, thanks.

**Pull request merged**